static/scripts/base.js

```diff
@@ -348,8 +348,27 @@ $(document).ready(() => {
 	});
 
 	// Window History Back
-	$('.historyback').on('click', () => {
-		const { search, origin } = window.location;
+	$('.historyback').on('click', async () => {
+		const { origin, href } = window.location;
+		const currentUrl = new URL(href);
+		const isCreatedSilently = currentUrl.searchParams?.get('isCreatedSilently');
+
+		// if homework was created silently, delete it
+		if (isCreatedSilently === 'true') {
+			const homeworkId = $('.section-upload').attr('data-parent-id');
+			const courseId = currentUrl.searchParams?.get('course');
+			const newReturnURL = courseId ? `/rooms/${courseId}` : '/tasks';
+
+			await $.ajax({
+				url: `/homework/${homeworkId}`,
+				type: 'DELETE',
+			});
+
+			currentUrl.searchParams?.set('returnUrl', newReturnURL);
+			window.history.pushState({}, null, currentUrl);
+		}
```

Comment:
Hey Leute, dass ist zu viel magic für ein use case da wird ihr euch selbst nach halbem Jahr fragen, wo was passiert. Bitte noch mal überdenken. 
https://www.askingbox.de/info/javascript-warnung-beim-verlassen-der-seite-anzeigen

Reply:
Ich find die Idee mit der Warnung auch ganz gut. Bei dem "beforeunload" gibt es aber wohl gewisse Kompatibilitätsprobleme: https://developer.mozilla.org/en-US/docs/Web/API/Window/beforeunload_event Ich hab das noch nie benutzt und weiß nicht, ob die für uns relevant wären.

Reply:
Haben den Code in das edit.js Script verschoben und unterbinden die Ausführung weitere Event Listeners.